### PR TITLE
Alerting: Support custom API URL for PagerDuty integration

### LINKED ISSUE #82741: Allow user to choose between Pagerduty global and Pagerduty EU endpoint
### Discussed in https://github.com/grafana/grafana/discussions/67187

<div type='discussions-op-text'>

<sup>Originally posted by **guusbeckett** April 25, 2023</sup>
At our company we use the PagerDuty EU endpoint, but it is not yet possible to use the EU endpoint of PagerDuty. 
As PagerDuty and PagerDuty EU are separate platforms, we cannot simply connect to the global endpoint and expect it to work.

I currently have a custom build that just changes the PagerDuty url to the EU one;
`https://events.pagerduty.com/v2/enqueue` becomes `https://events.eu.pagerduty.com/v2/enqueue`

Is it an idea to provide users with a new alert manager recipient and add _PagerDuty EU_ or would you prefer something else?

I'd be happy to implement this in the code.</div>

### PR DESCRIPTION
**Pull request by gaurav1999**
- Fixes: https://github.com/grafana/grafana/issues/82741
- Attached with https://github.com/grafana/alerting/pull/174

### CODE REVIEW

**Comment by gaurav1999** [comment]
cc @yuri-tceretian

File: pkg/services/ngalert/notifier/channels_config/available_channels.go [hunk 1]
```diff
@@ -363,6 +363,14 @@ func GetAvailableNotifiers() []*NotifierPlugin {
 					InputType:    InputTypeText,
 					PropertyName: "details",
 				},
+				{
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/notifier/channels_config/available_channels.go:366]: add comment `//New in 11.1`

File: pkg/services/ngalert/notifier/channels_config/available_channels.go [hunk 2]
```diff
@@ -363,6 +363,14 @@ func GetAvailableNotifiers() []*NotifierPlugin {
...
+					Label:        "API URL",
+					Description:  "API URL of your Pager Duty account",
+					Element:      ElementTypeInput,
+					InputType:    InputTypeText,
+					Placeholder:  "{{ .ExternalURL }}",
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/notifier/channels_config/available_channels.go:371]: this should be taken from `pagerduty.DefaultAPIURL`.

you can change it to the default URL and then update
once you import the new version of alerting module.

File: pkg/services/ngalert/notifier/channels_config/available_channels.go [hunk 3]
```diff
@@ -363,6 +363,14 @@ func GetAvailableNotifiers() []*NotifierPlugin {
 					InputType:    InputTypeText,
 					PropertyName: "details",
 				},
+				{
+					Label:        "API URL",
+					Description:  "API URL of your Pager Duty account",
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/notifier/channels_config/available_channels.go:368]: let's use the same description 

image

**Review by yuri-tceretian — COMMENTED** [review]
A few comments. 

Once the PR in alerting is merged, we need to update the reference to the new version of module.

File: pkg/services/ngalert/notifier/channels_config/available_channels.go (hunk 2 above)
Comment (gaurav1999) [reply pkg/services/ngalert/notifier/channels_config/available_channels.go:371]: If we are going with URL, then I think this prop should be DefaultURL, will be using that.

**Comment by yuri-tceretian** [comment]
Please run 
```
go get github.com/grafana/alerting@282da04de5edea3138b2d5180ec439251705eb57
```
in the root of this repository. It will import a version of alerting module that contains your changes. 
Please 
- commit `go.mod`  and `go.sum`
- in file `available_channels.go` add import `alertingPagerduty "github.com/grafana/alerting/receivers/pagerduty"` and update placeholder.

**Comment by gaurav1999** [comment]
> Please run
> 
> ```
> go get github.com/grafana/alerting@282da04de5edea3138b2d5180ec439251705eb57
> ```
> 
> in the root of this repository. It will import a version of alerting module that contains your changes. Please
> 
> * commit `go.mod`  and `go.sum`
> * in file `available_channels.go` add import `alertingPagerduty "github.com/grafana/alerting/receivers/pagerduty"` and update placeholder.

@yuri-tceretian Done from my side, thanks a ton!!

**Review by yuri-tceretian — APPROVED** [review]
LGTM

**Comment by yuri-tceretian** [comment]
Oh, CI fails because we forgot to change models for export:
- you need to add 
 a new field https://github.com/grafana/grafana/blob/cf601fab0906f1d811f1aa2fe18a67bc5421e3db/pkg/services/ngalert/api/tooling/definitions/contact_points.go#L137
Also, I would appreciate if you could update our terraform provider https://github.com/grafana/terraform-provider-grafana/blob/aaa20a20f309172c0b88071f80b9e9e5e52c75ae/internal/resources/grafana/resource_alerting_contact_point_notifiers.go#L1011

**Comment by yuri-tceretian** [comment]
@gaurav1999 it looks like you have committed from two accounts. Can you please make sure that you signed CLA with both accounts?

**Comment by gaurav1999** [comment]
> @gaurav1999 it looks like you have committed from two accounts. Can you please make sure that you signed CLA with both accounts?

Have reworked this, should be okay now, I have rebased it to my original author.

**Comment by yuri-tceretian** [comment]
> Have reworked this, should be okay now, I have rebased it to my original author.

**Comment by yuri-tceretian** [comment]
Nevermind, I squashed your commits into one. Should work now

**Comment by gaurav1999** [comment]
@yuri-tceretian Thanks a ton! , loved the experience

**Pull request merged**